modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -414,15 +468,75 @@ public void testWalCompressedBytes() throws Exception {
 
         stopAllGrids();
 
-        IgniteEx n1 = startGrid(0, (Consumer<IgniteConfiguration>)cfg ->
-            cfg.getDataStorageConfiguration().setWalCompactionEnabled(true));
+        IgniteEx n1 = startGrid(0,
+            (Consumer<IgniteConfiguration>)cfg -> cfg.getDataStorageConfiguration().setWalCompactionEnabled(true));
 
         n1.cluster().state(ACTIVE);
         awaitPartitionMapExchange();
 
         assertCorrectWalCompressedBytesMetrics(n1);
     }
 
+    /**
+     * Populates a cache w/32 KB of data
+     *
+     * @param igniteEx  Node.
+     */
+    private void populateCache(IgniteEx igniteEx) {
+        for (int i = 0; i < 10; i++)
+            igniteEx.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+    }
+
+    /**
+     * Check the state of wal archive, and whether the total size of
+     * wal (and possibly wal archive) match what is expected.
+     *
+     * @param igniteEx       Node.
+     * @param hasWalArchive  Whether wal archiving is enabled.
+     * @throws Exception     If failed.
+     */
+    private void checkWalArchiveAndTotalSize(IgniteEx igniteEx, boolean hasWalArchive) throws Exception {
+        FileWriteAheadLogManager walMgr = walMgr(igniteEx);
+
+        SegmentRouter router = walMgr.getSegmentRouter();
+
+        assertEquals(router.hasArchive(), hasWalArchive);
+
+        //wait to avoid race condition where new segments(and corresponding .tmp files) are created after totalSize has been calculated.
```

Comment:
`// Wait to avoid race condition where new segments(and corresponding .tmp files) are created after totalSize has been calculated.`